src/ragger/navigator/navigator.py

```diff
@@ -288,3 +288,70 @@ def navigate_until_snap(self,
         else:
             raise ValueError(f"Could not find first snapshot {first_golden_snap}")
         return img_idx
+
+    def navigate_until_text(self,
+                            navigate_instruction: NavIns,
+                            validation_instruction: NavIns,
+                            text: str,
+                            timeout: int = 30):
+        """
+        Navigate until some text is found on the screen content displayed.
+
+        This method may be left void on backends connecting to physical devices,
+        where a physical interaction must be performed instead.
+        This will prevent the instrumentation to fail (the void method won't
+        raise `NotImplementedError`), but the instrumentation flow will probably
+        get stuck (on further call to `receive` for instance) until the expected
+        action is performed on the device.
+
+        :param navigate_instruction: Navigation instruction to be performed until the text is found.
+        :type navigate_instruction: NavIns
+        :param validation_instruction: Navigation instruction to be performed once the text is found.
+        :type validation_instruction: NavIns
+        :param first_instruction_wait: Sleeping time before the first snapshot
+        :type first_instruction_wait: float
+        :param timeout: Timeout of the navigation loop if last snapshot is not found.
+        :type timeout: int
+
+        :return: img_idx
+        :rtype: int
+        """
+
+        if not isinstance(self._backend, SpeculosBackend):
+            # TODO remove this once the proper behavior of
+            # other backends is implemented.
+            return 0
+
+        start = time()
+
+        ctx = self._backend.wait_for_screen_change(2.0)
+        # Make sure to enter the navigation loop after
+        # there is a screen change (or wait 1 second).
+        # Useful when the navigate_until_text is used
+        # in transaction flows and the screen changes
+        # from the idle menu to a review start screen.
+        try:
+            ctx = self._backend.wait_for_screen_change(1.0, ctx)
+        except TimeoutError:
+            # Maybe the start screen of the transaction flow is
+            # already displayed so we'll wait 1 sec until
+            # the exception is thrown be we pass and go to the
```

Comment:
```suggestion
            # the exception is thrown then we pass and go to the
```
I guess there is a typo in there :)

Reply:
There is. Fixing it right now.